### ISSUE TITLE
TensorFlow dependence

### ISSUE THREAD

**Issue by zygmuntz** [rtavenar/tslearn#73]
You have a leaky abstraction there: tslearn uses Keras, but imports from TensorFlow directly:

```
c:\usr\anaconda\lib\site-packages\tslearn\shapelets.py in <module>()
     15 from keras.engine import InputSpec
     16 import numpy
---> 17 from tensorflow import set_random_seed
```

What if someone uses another backend?

**Comment by rtavenar** [comment]
You are right, and I have no solution for that at the moment (there does not seem to be a way to seed the RNG from `keras` directly), so I'll just add a dependence to tensorflow for the moment.

Thanks anyway.

**Comment by rtavenar** [comment]
Commited: 
https://github.com/rtavenar/tslearn/commit/4dbbf163b9aaa75005b20a02e07d5609d5a79c09
